doc/plugin_agent_svidstore_sconecas_sessionmanager.md

```diff
@@ -70,13 +73,17 @@ The placeholders needed by the plugin for each session are:
 |         `svid_session_template_file`      |              *             |                                 *                                 |
 |                     *                     |          `<\svid>`         |         SVID leaf certificate and its intermediates (PEM)         |
 |                     *                     |        `<\svid-key>`       |                           SVID key (PEM)                          |
-|                     *                     | `<\session-name-selector>` | Name of the session that will import the SVID and the private key |
-|                     *                     | `<\session-hash-selector>` | HASH of the session that will import the SVID and the private key |
+|                     *                     | `<\session-name-selector>` | Value of the `session_name` selector (Name of the session that will import the SVID and the private key) |
+|                     *                     | `<\session-hash-selector>` | Value of the `session_hash` selector (HASH of the session that will import the SVID and the private key) |
 |                     *                     |      `<\predecessor>`      |                       Sessions' predecessors                      |
 |       `bundle_session_template_file`      |              *             |                                 *                                 |
+|                     *                     |      `<\trust-bundle-session-name>`     |      Value of the `trust_bundle_session_name` selector                    |
 |                     *                     |      `<\trust-bundle>`     |                     Trust bundle for the SVIDs                    |
+|                     *                     |      `<\predecessor>`      |                       Sessions' predecessors                      |
 | `federated_bundles_session_template_file` |              *             |                                 *                                 |
+|                     *                     |      `<\fed-bundles-session-name>`     |      Value of the `fed_bundles_session_name` selector                    |
 |                                           |   `<\federated-bundles>`   |            Federated bundles associated with the SVIDs            |
+|                     *                     |      `<\predecessor>`      |                       Sessions' predecessors                      |
 
 **SVID session template examples** 
```

Comment:
On the template `spire-svid-<\session-name-selector>` we have an explicit secret (the X509 crt) with an implicit secret as a dependency. This isn't supported by CAS from what I could gather from my latest tests:

### CAS output:

```bash
Error: Failure during session preparation
Caused by: Parsing session from session template file 'session_custom.yaml' failed
Caused by: Validation of parsed session encountered problems
Caused by: Explicit value for secret 'svid' does not comply with the secret's description
Caused by: Failed to fetch private key 'svid_key'
Caused by: Failed to load value
Caused by: Secret was not found amongst explicit secret values, but it needs to be specified as it is a dependency of another explicit secret
```


### Session generated by SPIRE Agent

```bash
time="2021-11-11T13:22:43Z" level=info msg="Posting Session:" external=false name: spire-svid-benardis-totally-awesome-service2
version: "0.3"
predecessor: ~
secrets:
  - name: svid
    kind: x509
    value: |
        -----BEGIN CERTIFICATE-----
        MIIB6TCCAY+gAwIBAgIQTV0sQa0c26trhs91SKwFnDAKBggqhkjOPQQDAjAeMQsw
        CQYDVQQGEwJVUzEPMA0GA1UEChMGU1BJRkZFMB4XDTIxMTExMTEzMDc1NFoXDTIx
        MTExMTIzMDgwNFowHTELMAkGA1UEBhMCVVMxDjAMBgNVBAoTBVNQSVJFMFkwEwYH
        KoZIzj0CAQYIKoZIzj0DAQcDQgAEAQnuolqSwOmXTc8tNmQGVWeVXhY+8ZamDnnq
        uXSGiFQ+++ah6cyaU1lSZzyU/wAvR+aB2NF0dPjeWCTaKsLo3aOBrzCBrDAOBgNV
        HQ8BAf8EBAMCA6gwHQYDVR0lBBYwFAYIKwYBBQUHAwEGCCsGAQUFBwMCMAwGA1Ud
        EwEB/wQCMAAwHQYDVR0OBBYEFBb0csnVrkeUCQKAH3x92WIBj95oMB8GA1UdIwQY
        MBaAFKpT35dcreA5Wr1xuWedW0BRWvwMMC0GA1UdEQQmMCSGInNwaWZmZTovL2V4
        YW1wbGUub3JnL3Njb25lLXNlcnZpY2UwCgYIKoZIzj0EAwIDSAAwRQIgDz7JpsVc
        uSDyxwaIR5fXf6EUl6raXrkzjid1mGL5fe4CIQCHiSKYwjJ+TU75HXH1G8dcsAZY
        qo+zeKTQuP4ALi0UIQ==
        -----END CERTIFICATE-----
        -----BEGIN CERTIFICATE-----
        MIIB4DCCAWegAwIBAgIRAKzFF0P5DXQW2b8tHsPa2ncwCgYIKoZIzj0EAwMwHjEL
        MAkGA1UEBhMCVVMxDzANBgNVBAoMBlNQSUZGRTAeFw0yMTExMTEwMDM0MDJaFw0y
        MTExMTIwMDM0MTJaMB4xCzAJBgNVBAYTAlVTMQ8wDQYDVQQKEwZTUElGRkUwWTAT
        BgcqhkjOPQIBBggqhkjOPQMBBwNCAAQoVWuis8EEHtU4oOy3pgA8GI+5AQH1CvfW
        KLg0Ql3TVxXA0sTEsTk38REaGotaQwd5kOB7HcQMBv2BK05a9Uc9o4GFMIGCMA4G
        A1UdDwEB/wQEAwIBhjAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBSqU9+XXK3g
        OVq9cblnnVtAUVr8DDAfBgNVHSMEGDAWgBSHpfNXovA1rMD4ZMRU527TujnI6DAf
        BgNVHREEGDAWhhRzcGlmZmU6Ly9leGFtcGxlLm9yZzAKBggqhkjOPQQDAwNnADBk
        AjBiJ8o4/bcmNT93FYZBpQjZhuDy9MTYxbDQsfAxoiSXwoermcpBV7GkEsW9twVf
        rE4CMDU29Be7wANQ1GlmwycwJ62jxn6MQH7hX3Q6rCK1aF4OZvjRjv8zKvsbPDiy
        /33JZA==
        -----END CERTIFICATE-----

    export:
        session: benardis-totally-awesome-service2
        session_hash: 3cbc3afe9cb3b59502a37b1b8a1f5ac0d02f929bf3c6250a0b6fec5d3d6b3be4
    private_key: svid_key
  - name: svid_key
    kind: private-key
    export:
        session: benardis-totally-awesome-service2
="<unknown>" plugin_name=sconecas_sessionmanager plugin_type=SVIDStore subsystem_name=catalog
time="2021-11-11T13:22:43Z" level=error msg="cannot post SVID and its key into CAS " Post "https://5-4-0.scone-cas.cf:8081/session": x509: certificate signed by unknown authority="<unknown>" external=false plugin_name=sconecas_sessionmanager plugin_type=SVIDStore subsystem_name=catalog
```

Reply:
I've tested it with `5-4-0.scone-cas.cf` and `5-6-0.scone-cas.cf`.

Reply:
As we discussed, this is a configuration problem. The trust anchor is not set correctly. Because of this, I am going to mark this conversation as resolved.